src/libstore/derivations.cc

```diff
@@ -313,6 +313,15 @@ Derivation parseDerivation(const Store & store, std::string && s, std::string_vi
 }
 
 
+/**
+ * Print a derivation string literal to an std::string.
+ *
+ * This syntax does not generalize to the expression language, which needs to
+ * escape `$`.
+ *
+ * @param res Where to print to
+ * @param s Which logical string to print
+ */
```

Comment:
Maybe this function should be moved to `print.cc`?

Reply:
Which `print.cc`? This is for the aterm printing of dervations, and part of the store layer.

Reply:
I certainly wouldn't mind say having `derivation/{aterm,json}.{cc.hh}` to separate data types from serialization, and put both serializations on equal footing, but that seems like a separate PR to me.

Reply:
Not feasible, because that's a different library. We should keep these separate. As mentioned in the comment, this is a different function.
I can remove this mention of the language completely if we want to be very (overly?) thorough about libstore not depending on libexpr. Personally I think there's value in cross-referencing components in both directions, although this may not be the brightest example.